### PR TITLE
fix(stage-pages): initialize streaming speech provider config

### PR DESCRIPTION
**Pull request by leaft**
## Summary

- Wait for authentication before the streaming speech settings page requests the protected model catalog.
- Initialize the provider configuration before reading or writing its model.
- Apply the server availability and default model before loading model-specific voices.
- Add browser regressions for signed-out startup and the authentication transition.

## Root cause

Provider configurations are no longer all created during store startup. This page still assumed that its configuration existed after the model request completed, then read `config.model` from `undefined`. The model catalog response was not the cause of the error.

## Visual changes

| Before | After |
| --- | --- |
| The route shows an error boundary with the undefined model TypeError | The route shows the signed-out streaming speech provider settings |
| A direct signed-out visit requests the protected catalog and crashes. | A direct signed-out visit stays usable and asks the user to sign in. |

## Tests

- `pnpm exec vitest run src/pages/official-provider-speech-streaming.browser.test.ts` (from `apps/stage-web`)
- `pnpm typecheck`
- `pnpm lint`
- `pnpm -F @proj-airi/stage-web build`
- `sem diff upstream/main...HEAD --no-cosmetics -v --file-exts .ts .vue`

### CODE REVIEW

File: packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue [hunk 1]
```diff
@@ -65,24 +68,50 @@ async function loadVoices() {
...
+
+  await providersStore.initializeProvider(providerId)
+  if (!active)
+    return
+
   await providersStore.fetchModelsForProvider(providerId)
+  if (!active)
+    return
+
+  const available = getStreamingTtsAvailable()
```
[the remark this answers is not in the document]
Comment (leaft) [reply packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue]: Fixed in `3c7585f65`. `fetchModelsForProvider` now returns one serializable catalog with `models`, `available`, and `defaultModel`. The follower uses that action result instead of reading leader-local module state. I also added a leader/follower browser regression test for this boundary.

File: packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue [hunk 2]
```diff
@@ -65,24 +68,50 @@ async function loadVoices() {
...
+  providersStore.setProviderAvailabilityOverride(providerId, available)
+  if (!available) {
+    providersStore.setProviderUnconfigured(providerId)
+    return
+  }
+
+  providersStore.forceProviderConfigured(providerId)
+  streamingAvailable.value = true
```
[the remark this answers is not in the document]
Comment (leaft) [reply packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue]: Fixed in `3c7585f65`. The page now awaits the availability and provider-status actions before setting `streamingAvailable`. A regression test delays `forceProviderConfigured` and verifies that the voices request starts only after configuration completes.

File: packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue [hunk 3]
```diff
@@ -65,24 +68,55 @@ async function loadVoices() {
...
+  await providersStore.initializeProvider(providerId)
+  if (!active)
+    return
+
+  const catalog = await providersStore.fetchModelsForProvider(providerId)
+  if (!active)
+    return
+
+  const available = catalog.available === true
+  await providersStore.setProviderAvailabilityOverride(providerId, available)
```
[the remark this answers is not in the document]
Comment (leaft) [reply packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue]: Fixed in `306a4828b`. The page now treats an absent `available` value as an unknown discovery result. It preserves the existing availability override and provider status. An explicit `available: false` response still disables the provider. I added a browser regression test that returns HTTP 502 and checks that the configured state remains unchanged.

File: packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue [hunk 4]
```diff
@@ -65,24 +68,61 @@ async function loadVoices() {
...
+
+  const catalog = await providersStore.fetchModelsForProvider(providerId)
+  if (!active)
+    return
+
+  // An absent value means that discovery failed before the server returned an
+  // authoritative state. Keep the last configured state and availability
+  // override so a transient request failure cannot hide the provider.
+  if (catalog.available === undefined)
+    return
```
[the remark this answers is not in the document]
Comment (leaft) [reply packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue:91]: Fixed in follow-up PR #2445 (commit ef092ebc4). A failed catalog action now returns the last authoritative availability from its leader-owned state. The page restores that value and continues model-scoped voice loading while keeping the failed discovery distinct from an explicit available: false response. The HTTP 502 regression now verifies that the voices request still runs.

File: packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue [hunk 5]
```diff
@@ -65,24 +68,61 @@ async function loadVoices() {
...
+
+  streamingAvailable.value = true
+
+  // If the operator did not curate a default server-side, fall back to the
+  // first model in the same catalog response. Do not read synchronized model
+  // state here because its follower snapshot can arrive after the action.
+  serverDefaultModel.value = catalog.defaultModel ?? catalog.models[0]?.id ?? null
+  const config = providerConfig.value
+  if (config && !config.model && serverDefaultModel.value)
+    config.model = serverDefaultModel.value
```
[the remark this answers is not in the document]
Comment (leaft) [reply packages/stage-pages/src/pages/settings/providers/speech/official-provider-speech-streaming.vue:115]: Fixed in follow-up PR #2445 (commit ef092ebc4). Both the discovered default and picker updates now use awaited leader-owned actions in the provider-config store; the follower no longer mutates a replicated snapshot. The regression uses a real leader/follower runtime and verifies that the default is applied without replacing a later user selection.
